### PR TITLE
Apply google format & setup spotless Gradle plug-in

### PR DESCRIPTION
**Pull request by pepone**
This PR reformats Java source code using google-java-format, and adds the spotless Gradle plug-in to check the code format during build.

### CODE REVIEW

File: java/gradle/ice.gradle [hunk 1]
```diff
@@ -52,6 +54,15 @@ slice {
     }
 }
 
+spotless {
+  java {
+    target project.fileTree("src/main/java") {
```
Comment (externl) [inline java/gradle/ice.gradle:59]: Should we include the tests as well?

**Review by externl — APPROVED** [review]

File: java/gradle/ice.gradle (hunk 1 above)
Comment (pepone) [reply java/gradle/ice.gradle:59]: This is the project `src/main/java` directory. All projects including tests have this structure.

**Review by bernardnormier — APPROVED** [review]
Goin forward, how do you keep the code correctly formatted?

**Comment by pepone** [comment]
> Goin forward, how do you keep the code correctly formatted?

I added the spotless Gradle plug-in that checks the format. It checks the format with our regular build. The plug-in also includes a task for applying the required format fixes.

See https://github.com/diffplug/spotless

**Pull request merged**